pipelined/bevy_pbr2/src/render/pbr.wgsl

```diff
@@ -510,27 +533,31 @@ fn fragment(in: FragmentInput) -> [[location(0)]] vec4<f32> {
 
         var N: vec3<f32> = normalize(in.world_normal);
 
-        // FIXME: Normal maps need an additional vertex attribute and vertex stage output/fragment stage input
-        //        Just use a separate shader for lit with normal maps?
-        // #    ifdef STANDARDMATERIAL_NORMAL_MAP
-        //     vec3 T = normalize(v_WorldTangent.xyz);
-        //     vec3 B = cross(N, T) * v_WorldTangent.w;
-        // #    endif
+#ifdef VERTEX_TANGENTS
+#ifdef STANDARDMATERIAL_NORMAL_MAP
+        var T: vec3<f32> = normalize(in.world_tangent.xyz);
+        var B: vec3<f32> = cross(N, T) * in.world_tangent.w;
+#endif
+#endif
 
         if ((material.flags & STANDARD_MATERIAL_FLAGS_DOUBLE_SIDED_BIT) != 0u) {
             if (!in.is_front) {
                 N = -N;
+#ifdef VERTEX_TANGENTS
+#ifdef STANDARDMATERIAL_NORMAL_MAP
+                T = -T;
```

Comment:
We could avoid this extra `#ifdef` entirely if we move the normal flip right below the `var N` declaration.

Reply:
That is true, but then there would be more runtime logic. As such I chose to do it this way, prioritising runtime. Which way do you want?

Reply:
How is this:

```rust
        var N: vec3<f32> = normalize(in.world_normal);
        if ((material.flags & STANDARD_MATERIAL_FLAGS_DOUBLE_SIDED_BIT) != 0u) {
            if (!in.is_front) {
                N = -N;
            }
        }

#ifdef VERTEX_TANGENTS
#ifdef STANDARDMATERIAL_NORMAL_MAP
        var T: vec3<f32> = normalize(in.world_tangent.xyz);
        var B: vec3<f32> = cross(N, T) * in.world_tangent.w;
#endif
#endif
```

More runtime work than this?

```rust
        var N: vec3<f32> = normalize(in.world_normal);

#ifdef VERTEX_TANGENTS
#ifdef STANDARDMATERIAL_NORMAL_MAP
        var T: vec3<f32> = normalize(in.world_tangent.xyz);
        var B: vec3<f32> = cross(N, T) * in.world_tangent.w;
#endif
#endif

        if ((material.flags & STANDARD_MATERIAL_FLAGS_DOUBLE_SIDED_BIT) != 0u) {
            if (!in.is_front) {
                N = -N;
#ifdef VERTEX_TANGENTS
#ifdef STANDARDMATERIAL_NORMAL_MAP
                T = -T;
                B = -B;
#endif
#endif
            }
        }
```

To me this looks like _less_ overall work.

Reply:
Ah we don't flip `T` properly in that case.  Carry on